sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/tests/JobhostEndToEnd.cs

```diff
@@ -277,6 +333,122 @@ public void TestEventGridToCustomCollection_Batch([EventGridTrigger] ValidPoco[]
             }
         }
 
+        [Theory]
+        [TestCase("SingleEvent", "0")]
+        [TestCase("SingleEventString", "0")]
+        [TestCase("SingleEventJObject", "0")]
+        [TestCase("SingleReturnEvent", "0")]
+        // space separated string as event ids
+        [TestCase("ArrayEvent", "0 1 2 3 4")]
+        [TestCase("CollectorEvent", "0 1 2 3")]
+        [TestCase("AsyncCollectorEvent", "0 1 2 3 4 5 6")]
+        [TestCase("StringEvents", "0 1 2 3 4")]
+        [TestCase("JObjectEvents", "0 1 2 3 4")]
+        public async Task OutputCloudEventBindingParamsTests(string functionName, string expectedCollection)
+        {
+            List<CloudEvent> cloudEvents = new List<CloudEvent>();
+            List<EventGridEvent> egEvents = new List<EventGridEvent>();
+
+            Func<EventGridAttribute, IAsyncCollector<object>> eventConverter = (attr =>
+            {
+                var mockClient = new Mock<EventGridPublisherClient>();
```

Comment:
This test seems to test the cloud event only. Why do we need to mock both methods?

Reply:
There is another test like this one.